src/styles/blocks/stars.scss

```diff
@@ -0,0 +1,20 @@
+.stars {
+  display: flex;
+  flex-direction: row;
+
+  &__star {
+    background-image: url(../images/star.svg);
+    height: 16px;
+    width: 16px;
+    background-repeat: no-repeat;
+    background-position: center;
+
+    &:not(:last-child) {
+      margin-right: 4px;
+    }
+  }
+
+  &--4 .stars__star:nth-child(-n+4) {
```

Comment:
all modifiers should work (not just --4). And try to use [cycle](https://sass-lang.com/documentation/at-rules/control/for) for it